nsv13/code/modules/client/zoom.dm

```diff
@@ -1,11 +1,21 @@
 /client
-	var/current_view_size = 15
-	var/next_zoomout_time = 0
+	var/overmap_zoomout = 0
 
 /client/MouseWheel(src,delta_x,delta_y,location,control,params) //This lets you zoom in / out with your mouse
-	if(mob)
-		if(!mob.overmap_ship) //If theyre not in an overmap ship, don't let them zoom.
-			return ..()
+	if(!mob?.overmap_ship) //If they're not in an overmap ship, shortcut to standard behavior.
+		return ..()
+	var/list/modifier = params2list(params)
+	if(modifier["shift"])
+		if(delta_y > 0)
+			overmap_zoomout--
+		else
+			overmap_zoomout++
+
+
+		overmap_zoomout = CLAMP(overmap_zoomout, 0, 15)
+		rescale_view(overmap_zoomout, 0, ((40*2)+1)-15)
+
+/* //Oh my god KMC all of this is A W F U L
```

Comment:
I'm leaving this here, purely so it can be remembered how badly you over-engineered this shit.